### ISSUE TITLE
Google Cloud Function - Error: Could not load the default credentials. [URL is not defined]

### ISSUE THREAD

**Issue by catalin-nicolae-ciobanu** [googleapis/google-auth-library-nodejs#964]
I've noticed since 2 days ago that newly deployed GCF (deployed as Firebase Functions), as well as existing functions are consistently failing to authenticate (get a client). 

My code is adjusted according to the recommendation in [#798] and it was working until recently. It looks like this:

```typescript
import { GoogleAuth } from "google-auth-library";
import { AuthClient } from "google-auth-library/build/src/auth/authclient";

const googleAuth: GoogleAuth = new GoogleAuth({
    scopes: "https://www.googleapis.com/auth/pubsub"
});
let googleAuthClient: AuthClient;
```

later on, in a function's scope (not global) I do this:

```typescript
// (Lazy) Initialize Google Auth Library client
googleAuthClient = googleAuthClient || await googleAuth.getClient();
```

The error I get is:

> Stack:
(Error: Could not load the default credentials. Browse to https://cloud.google.com/docs/authentication/getting-started for more information.
    at GoogleAuth.getApplicationDefaultAsync (/srv/node_modules/google-auth-library/build/src/auth/googleauth.js:154:19)
    at <anonymous>
    at process._tickDomainCallback (internal/process/next_tick.js:229:7))

The error is always accompanied by a warning message:

> (node:2) MetadataLookupWarning: received unexpected error = URL is not defined code = UNKNOWN

#### Environment details

  - OS: the one used by GCF
  - Node.js version: 8
  - `google-auth-library` version: 6.0.0

I don't use a dedicated service account for this. So far I've always relied on Application Default Credentials.
Looking into the library's code it seems that there's something broken in the _is-running-inside-GCP_ detection mechanism. I've also checked the status of both firebase and google cloud and there seems to be no active issue related to cloud functions.

Thanks!

**Comment by bcoe** [comment]
@catalin-nicolae-ciobanu could you run on Node 10, which went GA this week. The `URL` class wasn't added until Node 10 (I believe you're running a version of `google-auth-library` which expects Node 10 or newer.

**Comment by catalin-nicolae-ciobanu** [comment]
Benjamin, thank you very much for your prompt reply!
You're right, that was the problem. Upgraded to nodejs 10 and everything is working just fine now (didn't know it went GA).

**Comment by bcoe** [comment]
@catalin-nicolae-ciobanu awesome 🎉 don't hesitate to follow up if you bump into any more issues.

**Comment by jemerson3** [comment]
I'm running node v12.16.2 and I'm still getting this error.

**Comment by martijnvdbrug** [comment]
> I'm running node v12.16.2 and I'm still getting this error.

This question is about Google Cloud Function, which doesn't have a Node 12 runtime yet.

**Comment by lawonga** [comment]
TL;DR

^ To everyone else who has this error, DOUBLE CHECK your node version!

**Node 8 got deprecated June 5, 2020.**

**Comment by philipperocha** [comment]
How to update Node to v10 for GCF??

**Comment by philipperocha** [comment]
> I'm running node v12.16.2 and I'm still getting this error.

I found it by myself, but I could not by browsing on google.

Go To packages.json, and change "engines":{ "node"}

  "engines": {
    "node": "10"
  },

Problem Solved!!
